File: lib/data/services/ws/ws_types.dart
enum WsMessageType {
  register('register'),
  challenge('challenge'),
  challengeResponse('challengeResponse'),
  registered('registered'),
  sessionInit('session-init'),
  sessionAck('session-ack'),
  sessionEnd('session-end'),
  rpc('rpc'),
  response('response'),
  event('event'),
  error('error');

  const WsMessageType(this.value);
  final String value;

  static WsMessageType? fromValue(String? raw) {
    if (raw == null) return null;
    for (final type in WsMessageType.values) {
      if (type.value == raw) return type;
    }
    return null;
  }
}

enum WsRole {
  client('client'),
  machine('machine');

  const WsRole(this.value);
  final String value;
}

File: internal/service/ws_types.go
package service

// WSType defines the WebSocket message type for the relay protocol.
type WSType string

const (
	WSTypeRegister          WSType = "register"
	WSTypeChallenge         WSType = "challenge"
	WSTypeChallengeResponse WSType = "challengeResponse"
	WSTypeRegistered        WSType = "registered"
	WSTypeSessionInit       WSType = "session-init"
	WSTypeSessionAck        WSType = "session-ack"
	WSTypeSessionEnd        WSType = "session-end"
	WSTypeRPC               WSType = "rpc"
	WSTypeResponse          WSType = "response"
	WSTypeEvent             WSType = "event"
	WSTypeError             WSType = "error"
)

// WSRole defines the registering WebSocket client role.
type WSRole string

const (
	WSRoleClient  WSRole = "client"
	WSRoleMachine WSRole = "machine"
)
